R/parameters.R

```diff
@@ -13,31 +13,61 @@
 ##' 
 ##' @export
 parameters_demographic <- function() {
+  
   age_bins <- get_age_bins()
 
   ## Set up population denominators
   country <- "Democratic Republic of Congo"
   data <- squire::get_population(country)
   N_age <- c(data$n[1:15], sum(data$n[16:17])) # combine 75+
+  names(N_age) <- age_bins$label
 
-  ## Add in Sex workers (SW) and People who Buy Sex (PBS)
-  idx_15_49  <- age_bins$start >= 15 & age_bins$end < 49
-  N_15_49 <- N_age * idx_15_49
-
-  p_SW <- 0.007 * 0.5 # 0.7% women (50%) 15-49 Laga et al
-  N_SW <- round(p_SW * N_15_49)
+  ## Add in (child/adult) sex workers, (C/ASW), people who buy sex (PBS),
+  ## and healthcare workers (HCW)
+  
+  ## CSW: age 12-17
+  w_12_17 <- proportion_in_age_bins(12, 17)
+  # 60% 10-14 + 60% 15-19
+  N_12_17 <- N_age * w_12_17
+  
+  ## ASW: age 18-49
+  w_18_49  <- proportion_in_age_bins(18, 49)
+  N_18_49 <- N_age * w_18_49
 
+  p_SW <- 0.007 * 0.5 # 0.7% women (50%) 15-49 Laga et al - assume this holds down to age 12
+  N_CSW <- round(p_SW * N_12_17)
+  N_ASW <- round(p_SW * N_18_49)
+  
+  ## HCW / PBS: age 20-49
+  w_20_49 <- proportion_in_age_bins(20, 49)
+  N_20_49 <- N_age * w_20_49
+  
+  ## PBS
+  # not changing this based on age as we only heard about young SWs not young PBS
   p_PBS <- 0.11 * 0.5 # 11% men (50%) 15-49 DHS https://www.statcompiler.com/en/
-  N_PBS <- round(p_PBS * N_15_49)
-
-  N <- c(N_age - N_SW - N_PBS, sum(N_SW), sum(N_PBS))
+  # we assume all greater than 20 to avoid vaccine issues
+  N_PBS <- round(p_PBS * N_20_49)
+  
+  
+  ## HCW
+  p_HCW <- 0 # for now
+  N_HCW <- round(p_HCW, N_20_49)
+
+  N <- c(N_age - N_ASW - N_PBS - N_CSW - N_HCW,
+         CSW = sum(N_CSW),
+         ASW = sum(N_ASW),
+         PBS = sum(N_PBS),
+         HCW = sum(N_HCW))
+  
+  # check allocated properly
+  sum(N) - sum(N_age)
```

Comment:
Did you want to leave this line in?

Reply:
There's a couple of other similar checks - do we want to leave these as is or perhaps comment them?